src/fireedge/src/modules/utils/schema.js

```diff
@@ -0,0 +1,627 @@
+ */
+
+/**
+ * @callback CreateStepsCallback
+ * @param {object} stepProps - Properties passes to all Step functions
+ * @param {object} initialValues - Initial values to form
+ * @returns {StepsForm & ExtraParams}
+ */
+
+/**
+ * @callback CreateFormCallback
+ * @param {object} props - Properties passes to schema and field
+ * @param {object} initialValues - Initial values to form
+ * @returns {Form & ExtraParams}
+ */
+
+// ----------------------------------------------------------
+// Constants
+// ----------------------------------------------------------
+
+/** @enum {Function} Sorters */
+export const OPTION_SORTERS = {
+  default: (a, b) => `${a.value}`.localeCompare(`${b.value}`),
+  numeric: (a, b) =>
+    `${a.value}`.localeCompare(`${b.value}`, undefined, {
+      numeric: true,
+      ignorePunctuation: true,
+    }),
+  unsort: () => null,
+}
+
+const SEMICOLON_CHAR = ';'
+
+const requiredSchema = (mandatory, schema) =>
+  mandatory ? schema.required() : schema.notRequired().nullable()
+
+const getRange = (options) => options?.split?.('..').map(parseFloat)
+
+const getValuesFromArray = (options, separator = SEMICOLON_CHAR) =>
+  options?.split(separator)
+
+const getOptionsFromList = (options = [], sorter) => {
+  const config = { addEmpty: false }
+  sorter && (config.sorter = sorter)
+
+  return arrayToOptions([...new Set(options)], config)
+}
+
+const parseUserInputValue = (value) => {
+  if (value === true) {
+    return 'YES'
+  } else if (value === false) {
+    return 'NO'
+  } else if (
+    Array.isArray(value) &&
+    value.every((v) => typeof v === 'string')
+  ) {
+    return value.join(',')
+  } else return value
+}
+
+// ----------------------------------------------------------
+// Function (to export)
+// ----------------------------------------------------------
+
+/**
+ * Get field properties to represent an user input defined by OpenNebula.
+ *
+ * @param {UserInputObject} userInput - User input from OpenNebula document
+ * @returns {Field} Field properties
+ */
+export const schemaUserInput = ({
+  mandatory,
+  type,
+  min,
+  max,
+  options,
+  default: defaultValue,
+  sorter,
+}) => {
+  switch (type) {
+    case USER_INPUT_TYPES.fixed: {
+      const isNumeric = !isNaN(defaultValue)
+      const ensuredValue = isNumeric ? parseFloat(defaultValue) : defaultValue
+      const validation = isNumeric ? number() : string().trim()
+
+      return {
+        type: INPUT_TYPES.TEXT,
+        htmlType: isNaN(+defaultValue) ? 'text' : 'number',
+        validation: validation
+          .default(ensuredValue)
+          // ensures to send the value
+          .afterSubmit(() => defaultValue),
+        fieldProps: { disabled: true },
+        readOnly: true,
+      }
+    }
+    case USER_INPUT_TYPES.text:
+    case USER_INPUT_TYPES.text64:
+    case USER_INPUT_TYPES.password:
+      return {
+        type: INPUT_TYPES.TEXT,
+        htmlType: type === USER_INPUT_TYPES.password ? 'password' : 'text',
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears your code may contain a hardcoded secret. We recommend replacing it with AWS Secrets Manager references to enhance security and follow best practices. For more information, please refer [OWASP password storage cheat sheet](https://www.owasp.org/index.php/Password_Storage_Cheat_Sheet).